docs/SPEC.md

```diff
@@ -839,6 +833,32 @@ Side effects can occur in two ways.
 
 A function produces side effects when it is explicitly declared to have side effects or when it calls a function that itself produces side effects.
 
+## Package initialization
+
+A package with no imports is initialized by first initializing and assigning all declared options and then subsequently initializing and assigning all declared variables.
+In particular all package options are initialized before any variables.
+Options cannot depend on any variables declared in the same package block.
+If a package has imports, all imported packages are initialized before any options or variables.
+
+### Variable initialization
+
+Variables are initialized in declaration order but only after all of the variables they depend on have been initialized.
```

Comment:
I think this should just say the variables initialization and declaration order is not important. The spec does not need to state how we accomplish it.